File: Arrays/Easy/C++/1832. Check if the Sentence Is Pangram.cpp
class Solution {
public:
    bool checkIfPangram(string sentence) {

        vector<int> arr(26, 0);

        for(int i = 0; i < sentence.length(); i++){
            arr[sentence.at(i) - 'a'] = 1;
        }

        for(int i = 0; i < 26; i++){
            if(arr[i] == 0){
                return false;
            }
        }

        return true;


        
        
        // set<char> s;

        // for(int i = 0; i < sentence.length(); i++){
        //     s.insert(sentence.at(i));
        // }

        // return s.size() == 26;
    }
};

File: Strings/Easy/1832. Check if the Sentence Is Pangram.java
class Solution {
    public boolean checkIfPangram(String sentence) {

        // Time Complexity : O(n + 26)

        // Space Complexity : O(26)

        
        int[] freq = new int[26];

        for(int i = 0; i < sentence.length(); i++){
            char ch = sentence.charAt(i);
            freq[ch - 'a']++;
        }

        for(Integer i: freq){
            if(i == 0){
                return false;
            }
        }

        return true;
    }
}
